akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala

```diff
@@ -390,5 +393,13 @@ private[akka] final class TestProbeImpl[M](name: String, system: ActorSystem[_])
     testActor.asInstanceOf[ActorRef[AnyRef]] ! Stop
   }
 
+  def tell(m: M) = testActor.tell(m)
+
+  // impl InternalRecipientRef, ask isn't supported
+  def provider: ActorRefProvider = throw new UnsupportedOperationException("ask pattern not supported")
```

Comment:
Can't it return the provider from system here (probably requires a cast to `ExtendedActorSystem`) and that'd make ask work as well?

Reply:
Does it even make sense to ask a probe, though?

I guess one could expect the ask and have the test case reply, but it seems to me that the best practice would be to use a probe only to ensure that the ask is made and to use an actual `Behavior` to test the entire request/response interaction (otherwise you're introducing some risk of mock-skew).

Reply:
I was thinking about the use case of passing the probe to some actor that will then ask it, not asking it from the test. But maybe that is hypothetic because you'd probably have an ActorRef and use .ref in that case anyway.

Reply:
I agree that we can use the provider from the ActorSystem so that the probe works in the same way as the probe.ref.